bootstrapper/internal/kubernetes/kubernetes.go

```diff
@@ -186,10 +186,29 @@ func (k *KubeWrapper) InitCluster(
 		return nil, fmt.Errorf("unmarshalling helm releases: %w", err)
 	}
 
+	log.Infof("Installing Cilium")
 	if err = k.helmClient.InstallCilium(ctx, k.client, helmReleases.Cilium, setupPodNetworkInput); err != nil {
 		return nil, fmt.Errorf("installing pod network: %w", err)
 	}
 
+	log.Infof("Waiting for Cilium to become healthy")
+	timeToStartWaiting := time.Now()
+	// TODO(Nirusu): Reduce the timeout when we switched the package repository - this is only this high because I once
+	// saw polling times of ~16 minutes when hitting a slow PoP from Fastly (GitHub's / ghcr.io CDN).
+	waitCtx, cancel = context.WithTimeout(ctx, 20*time.Minute)
+	defer cancel()
+	if err := k.clusterUtil.WaitForCilium(waitCtx, log); err != nil {
+		return nil, fmt.Errorf("waiting for Cilium to become healthy: %w", err)
+	}
+	timeUntilFinishedWaiting := time.Since(timeToStartWaiting)
+	log.With(zap.Duration("duration", timeUntilFinishedWaiting)).Infof("Cilium became healthy")
+
+	log.Infof("Restart Cilium")
+	if err := k.clusterUtil.FixCilium(ctx); err != nil {
+		log.With(zap.Error(err), "FixCilium failed", zap.Error(err))
+		// Continue and don't throw an error here - things might be okay.
```

Comment:
Could you explain why things might still be ok if we see an error? And why we throw an error even if we want to continue in that case?

Reply:
>Could you explain why things might still be ok if we see an error?
The `FixCilium` function only restarts Cilium the first time to work around some Pod2Pod networking bug. However, given that it seemed to mostly work fine in the old version where we only did this after we installed all other Helm charts, it will likely be fine if this fails (for whatever reason) and the rest of the init likely is still able to go through. 

Worst case if the Pod2Pod networking bug still exists but everything else was set up correctly, you just restart Cilium or the node after the init.

>And why we throw an error even if we want to continue in that case?

For the logs. The old versions passed the logger down, but why do that when we can just return the error and let the caller choose to handle this? Seems more generalized and straightforward to me.

Reply:
Speaking purely from a UX perspective: it seems less than ideal to me to force the user to go in and restart pods after we have "Successfully initiated" a cluster. If I understand you correctly the more acurate CLI output would be sth like: "We may have successfully initiated a cluster, but maybe it broke, so you better go back to check and fix those Cilium pods"  :D

If pod2pod networking is broken until `FixCilium` completed successfully, why not retry that function?

Reply:
>If pod2pod networking is broken until FixCilium completed successfully, why not retry that function?

1. We don't know for sure we into the bug.
2. Because the logic itself rarely fails. Looking in OpenSearch, we have only ~40 fails and most of them are where the context deadline exceeded. All remaining issues, except one were "container is already in removing state" which should be okay I guess. The last one was "container is in starting state, can't be removed" which shouldn't have happened due to the wait, but yeah, this _very_ rare.

Let's assume we try. How often should we try? Should we block & wait? Should we do this in a separate goroutine? There's not even a guarantee we ran into the bug.

>If I understand you correctly the more acurate CLI output would be sth like: "We may have successfully initiated a cluster, but maybe it broke, so you better go back to check and fix those Cilium pods" :D

But it's not permanently broken. If we hard fail (which we also did not before, so it's not a regression but rather could introduce a regression), why would you force the user to recreate their whole infrastructure? 

Generally just printing a warning to the CLI would be fine here (and better than `terminate` and `create`, but we don't have that mechanism unfourtunaly and would require larger refactors to make the bootstrapped init process a multi-message gRPC.

Reply:
Oh. I didn't realize that FixCilium also didn't break init in case it failed. Nvm then :D

Alright, lets see where this goes.

Reply:
To be honest I was surprised init even worked this well before despite `FixCilium` only being called at the end (since IIRC the cert-manager check is Pod2Pod networking) :D